packages/plugin-ext-vscode/src/node/plugin-vscode-file-handler.ts

```diff
@@ -50,34 +39,45 @@ export class PluginVsCodeFileHandler implements PluginDeployerFileHandler {
         });
     }
 
+    protected getNormalizedExtensionId(pluginId: string): string {
+        return filenamify(pluginId, { replacement: '_' }).replace(/\.vsix$/, '');
+    }
+
     async handle(context: PluginDeployerFileHandlerContext): Promise<void> {
-        const id = context.pluginEntry().id();
-        const extensionDir = await this.getExtensionDir(context);
-        console.log(`[${id}]: trying to decompress into "${extensionDir}"...`);
-        if (context.pluginEntry().type === PluginType.User && await fs.pathExists(extensionDir)) {
-            console.log(`[${id}]: already found`);
-            context.pluginEntry().updatePath(extensionDir);
+        const id = this.getNormalizedExtensionId(context.pluginEntry().id());
+        const extensionDeploymentDir = await this.getExtensionDeploymentDir(id);
+        try {
+            console.log(`[${id}]: trying to decompress into "${extensionDeploymentDir}"...`);
+
+            if (await fs.pathExists(extensionDeploymentDir)) {
+                console.log(`[${id}]: deployment dir "${extensionDeploymentDir}" already exists ...`);
+                context.pluginEntry().updatePath(extensionDeploymentDir);
+                return;
+            }
+
+            if (!await decompressExtension(context.pluginEntry().path(), extensionDeploymentDir)) {
+                console.error(`[${id}]: decompression failed`);
+                return;
+            }
+
+            console.log(`[${id}]: decompressed`);
+        } catch (error) {
+            console.error(`[${id}]: error while decompressing`, error);
```

Comment:
Why not let the error propagate? How is the invoker going to tell sometihing failed?